Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -838,6 +1097,293 @@ void MatchTOF::doMatching(int sec)
+    trkLTInt.clear();
+    trkLTInt.reserve(BCcand.size());
+    deltaPos.clear();
+    deltaPos.reserve(BCcand.size());
+    nStepsInsideSameStrip.clear();
+    nStepsInsideSameStrip.reserve(BCcand.size());
+
+    // printf("%d) ts_error = %f -- z_error = %f\n", itrk, trackWork.second.getTimeStampError(), trackWork.second.getTimeStampError() * vdrift);
+
+    //    Printf("intLT (before doing anything): length = %f, time (Pion) = %f", intLT.getL(), intLT.getTOF(o2::track::PID::Pion));
+    int istep = 1;    // number of steps
+    float step = 1.0; // step size in cm
+                      //uncomment for local debug
+                      /*
+    //trefTrk.getXYZGlo(posBeforeProp);
+    //float posBeforeProp[3] = {trefTrk.getX(), trefTrk.getY(), trefTrk.getZ()}; // in local ref system
+    //printf("Global coordinates: posBeforeProp[0] = %f, posBeforeProp[1] = %f, posBeforeProp[2] = %f\n", posBeforeProp[0], posBeforeProp[1], posBeforeProp[2]);
+    //Printf("Radius xy = %f", TMath::Sqrt(posBeforeProp[0]*posBeforeProp[0] + posBeforeProp[1]*posBeforeProp[1]));
+    //Printf("Radius xyz = %f", TMath::Sqrt(posBeforeProp[0]*posBeforeProp[0] + posBeforeProp[1]*posBeforeProp[1] + posBeforeProp[2]*posBeforeProp[2]));
+    */
+
+#ifdef _ALLOW_TOF_DEBUG_
+    if (mDBGFlags) {
+      (*mDBGOut) << "propOK"
+                 << "track=" << trefTrk << "\n";
+    }
+#endif
+
+    int detIdTemp[5] = {-1, -1, -1, -1, -1}; // TOF detector id at the current propagation point
+
+    double reachedPoint = mXRef + istep * step;
+
+    // initializing
+    for (int ibc = 0; ibc < BCcand.size(); ibc++) {
+      for (int ii = 0; ii < 2; ii++) {
+        nStepsInsideSameStrip[ibc][ii] = 0;
+        for (int iii = 0; iii < 5; iii++) {
+          detId[ibc][ii][iii] = -1;
+        }
+      }
+    }
+    while (propagateToRefX(trefTrk, reachedPoint, step, intLT) && reachedPoint < Geo::RMAX) {
+      // while (o2::base::Propagator::Instance()->PropagateToXBxByBz(trefTrk,  mXRef + istep * step, o2::constants::physics::MassPionCharged, MAXSNP, step, 1, &intLT) && nStripsCrossedInPropagation <= 2 && mXRef + istep * step < Geo::RMAX) {
+
+      trefTrk.getXYZGlo(pos);
+      for (int ii = 0; ii < 3; ii++) { // we need to change the type...
+        posFloat[ii] = pos[ii];
+      }
+      // uncomment below only for local debug; this will produce A LOT of output - one print per propagation step
+      /*
+	Printf("posFloat[0] = %f, posFloat[1] = %f, posFloat[2] = %f", posFloat[0], posFloat[1], posFloat[2]);
+	Printf("radius xy = %f", TMath::Sqrt(posFloat[0]*posFloat[0] + posFloat[1]*posFloat[1]));
+	Printf("radius xyz = %f", TMath::Sqrt(posFloat[0]*posFloat[0] + posFloat[1]*posFloat[1] + posFloat[2]*posFloat[2]));
+      */
+
+      reachedPoint += step;
+
+      // check if you fall in a strip
+      for (int ibc = 0; ibc < BCcand.size(); ibc++) {
+        for (int idet = 0; idet < 5; idet++)
+          detIdTemp[idet] = -1;
+
+        if (side > 0)
```

Comment:
I would suggest to define the side as
````
int side = track.hasASideClustersOnly() ? 1 : (track.hasCSideClustersOnly() ? -1 : 0);
````
Then, if the ``side==0`` (i.e. the track crosses CE), then you should move the track, its time0 is already constrained.

Reply:
done